WordPress/Classes/ViewRelated/Jetpack/Branding/Button/CustomImageBackgroundButton.swift

```diff
@@ -0,0 +1,41 @@
+import UIKit
+
+/// Use this UIButton subclass to set a custom background for the button image, different from tintColor and backgroundColor.
+/// If there's not image, it has no effect.
+class CustomImageBackgroundButton: UIButton {
+
+    private lazy var imageBackgroundView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+
+    /// Sets a custom circular background color below the imageView, different from the button background or tint color
+    /// - Parameters:
+    ///   - color: the custom background color
+    ///   - ratio: the extent of the background view that lays below the button image view (default: 0.75 of the image view)
+    func setImageBackgroundColor(_ color: UIColor, ratio: CGFloat = 0.75) {
+        guard let imageView = imageView else {
+            return
+        }
+        imageBackgroundView.backgroundColor = color
+        insertSubview(imageBackgroundView, belowSubview: imageView)
+        imageBackgroundView.clipsToBounds = true
+        imageBackgroundView.isUserInteractionEnabled = false
+        NSLayoutConstraint.activate([
+            imageBackgroundView.centerXAnchor.constraint(equalTo: imageView.centerXAnchor),
+            imageBackgroundView.centerYAnchor.constraint(equalTo: imageView.centerYAnchor),
+            imageBackgroundView.heightAnchor.constraint(equalTo: imageView.widthAnchor, multiplier: ratio),
+            imageBackgroundView.widthAnchor.constraint(equalTo: imageView.widthAnchor, multiplier: ratio),
+        ])
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        guard imageView != nil else {
+            return
+        }
+        imageBackgroundView.layer.cornerRadius = imageBackgroundView.frame.height / 2
```

Comment:
👋 @Gio2018 - will this mean that all instances of `CustomImageBackgroundButton` will have a round background view?

Reply:
A rounded image with a rounded background, since this is the only case that we are handling at the moment. We could generalize it if it comes to more general image shapes in the future.

Reply:
> since this is the only case that we are handling at the moment

How would you feel about making `class CustomImageBackgroundButton: UIButton {` have a less generic name if it doesn't currently support generic use cases?

Reply:
Sure: how about one of these?

- `CircularImageWithCustomBackgroundButton` 
- `ColorCircularImageButton`
- `CircularImageButton` (the description will explain that the background is customizable)
- `JetpackBrandingBaseButton` (since it's used for jetpack branding)
- Something else?

Reply:
Any of those LGTM! Especially the circular ones. Thanks @Gio2018.

Reply:
done!

Reply:
Thanks!